### PR TITLE
easteregg/fix-additional-digit

### PR DESCRIPTION
**Pull request by easteregg**

### CODE REVIEW

File: src/javascript/app/pages/trade/digit_trade.js [hunk 1]
```diff
@@ -156,9 +160,10 @@ const DigitDisplay = (() => {
                 start        : contract.entry_tick_time,
                 end          : contract.exit_tick_time,
             };
-
-            // force rerender the table by sending the history
-            BinarySocket.send(request, { callback: redrawFromHistory });
+            if (is_redraw_possible) {
+                // force rerender the table by sending the history
+                BinarySocket.send(request, { callback: redrawFromHistory });
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: will this cause multiple redraws on every tick received?

Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: Per slack talk, this won't redraw on every tick, only conditionally on contract end.

File: src/javascript/app/pages/trade/digit_ticker.js [hunk 2]
```diff
@@ -177,9 +177,9 @@ const DigitTicker = (() => {
     };
 
     const setElements = () => {
-        el_peek     = el_container.querySelector('.peek');
-        el_peek_box = el_container.querySelector('.peek-box');
-        el_mask     = el_peek_box.querySelector('.peek-box > .mask');
+        el_peek     = el_container ? el_container.querySelector('.peek') : null;
+        el_peek_box = el_peek ? el_container.querySelector('.peek-box') : null;
+        el_mask     = el_peek_box ? el_peek_box.querySelector('.peek-box > .mask') : null;
```
Comment (easteregg) [inline src/javascript/app/pages/trade/digit_ticker.js]: This is to prevent trackjs errors

Comment (negarn) [reply src/javascript/app/pages/trade/digit_ticker.js]: shouldn't it be `el_peek_box.querySelector('> .mask')`? is there another peek box inside the peek box?

File: src/javascript/app/pages/trade/digit_trade.js [hunk 3]
```diff
@@ -13,13 +13,18 @@ const DigitDisplay = (() => {
         tick_count,
         spot_times;
 
+    // Subscribe if contract is still ongoing/running.
     const subscribe = (request) => {
-        // Subscribe if contract is still ongoing/running.
-        if (contract.current_spot_time < contract.date_expiry) {
+        request.end       = 'latest';
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: extra space before `=`

File: src/javascript/app/pages/trade/digit_trade.js [hunk 4]
```diff
@@ -13,13 +13,18 @@ const DigitDisplay = (() => {
...
+        if (contract.exit_tick_time) {
+            request.end = +contract.exit_tick_time;
+            request.count = +contract.tick_count;
+            if (+contract.tick_count === 1) {
+                request.end += 1; // TODO: backend sends the improper response when end and start are the same for 1 tick contracts. remove this block on fix
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: please refer to it as `API` instead

File: src/javascript/app/pages/trade/digit_trade.js [hunk 5]
```diff
@@ -81,42 +87,11 @@ const DigitDisplay = (() => {
             tick_count,
             {
                 quote: contract.status !== 'open' ? contract.exit_tick : spot,
-                epoch: contract.status !== 'open' ? contract.exit_tick_time : contract.date_expiry,
+                epoch: +contract.exit_tick_time || contract.current_spot_time,
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: shouldn't we also use `+` on `current_spot_time` for consistency/assurance?

File: src/javascript/app/pages/trade/digit_trade.js [hunk 6]
```diff
@@ -46,9 +51,10 @@ const DigitDisplay = (() => {
             contract.tick_count,
             contract.status
         );
+
         const request = {
             ticks_history: contract.underlying,
-            start        : contract.date_start,
+            start        : contract.entry_tick_time,
```
Comment (negarn) [inline src/javascript/app/pages/trade/digit_trade.js]: perhaps `+` here too

File: src/javascript/app/pages/trade/digit_trade.js (hunk 4 above)
Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: Sure.

File: src/javascript/app/pages/trade/digit_trade.js (hunk 5 above)
Comment (easteregg) [reply src/javascript/app/pages/trade/digit_trade.js]: Yes we should.

**Pull request merged**